src/service/repo/signedMessageRepo.ts

```diff
@@ -90,29 +91,40 @@ async function sendSignedMessage(
 }
 
 async function getAllSignedChannels() {
+  const timeStamp = StorageUtils.channelSignedTimeStamps.get();
+  const url = timeStamp
+    ? `${baseUrl.getAllSignedChannels}?last_fetch_date=${timeStamp}`
+    : baseUrl.getAllSignedChannels;
   try {
-    const response = await api.get(baseUrl.getAllSignedChannels);
-
+    const response = await api.get(url);
     if (response.status === 200) {
+      console.log('SUCCESS URL', url);
       return Promise.resolve(response.data?.data);
     }
 
     return Promise.reject(response.data?.status);
   } catch (e) {
+    console.log('FAILED URL', url);
     console.log(e);
     return Promise.reject(e);
   }
 }
 
 async function getAllSignedPostNotifications() {
+  const timeStamp = StorageUtils.channelAnonTimeStamps.get();
+  const url = timeStamp
+    ? `${baseUrl.getAllSignedPostNotifications}?last_fetch_date=${timeStamp}`
+    : baseUrl.getAllSignedPostNotifications;
   try {
-    const response = await api.get(baseUrl.getAllSignedPostNotifications);
+    const response = await api.get(url);
     if (response.status === 200) {
+      console.log('SUCCESS URL', url);
       return Promise.resolve(response.data?.data);
     }
 
     return Promise.reject(response.data?.status);
   } catch (e) {
+    console.log('FAILED URL', url);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Similar to the previous function, `getAllSignedPostNotifications()` now fetches data based on timestamps. The same suggestion for improved error handling applies here.
```diff
+ async function getAllSignedPostNotifications() {
+   const timeStamp = StorageUtils.channelAnonTimeStamps.get();
+   const url = timeStamp
+     ? `${baseUrl.getAllSignedPostNotifications}?last_fetch_date=${timeStamp}`
+     : baseUrl.getAllSignedPostNotifications;
+    try {
+     const response = await api.get(url);
+      if (response.status === 200) {
+       console.log('SUCCESS URL', url);
+        return Promise.resolve(response.data?.data);
+      }
+  
+      return Promise.reject(new Error(`Failed to fetch signed post notifications with status: ${response.data?.status}`));
+    } catch (e) {
+     console.error('FAILED URL', url);
+      console.error(e);
+      return Promise.reject(new Error(`Failed to fetch signed post notifications with error: ${e.message}`));
+    }
+  }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->